packages/blade/src/components/Menu/_decisions/decisions.md

```diff
@@ -0,0 +1,372 @@
+   */
+  isOpen?: boolean;
+
+  /**
+   * On Menu open change callback
+   */
+  onOpenChange?: ({ isOpen }: { isOpen: boolean }) => void;
+
+  /**
+   * Should menu open on click or hover
+   *
+   * @default 'click'
+   */
+  openInteraction?: 'hover' | 'click';
+};
+```
+
+### MenuOverlay
+
+This overlay will be flexible overlay which can have any component. Static / Non-Static / MenuItem / etc
+
+```ts
+type MenuOverlay = {
+  /**
+   * A slot inside Menu's overlay.
+   *
+   * Supports MenuItem or any other component
+   */
+  children: React.ReactElement[];
+};
+```
+
+### MenuItem
+
+```ts
+type MenuItemProps = {
+  /**
+   * title of item
+   */
+  title: string;
+
+  /**
+   * Description text for the item
+   */
+  description?: string;
+
+  /**
+   * Click handler for MenuItem
+   *
+   * Absense of this prop and href will turn the item into non-interactive item
+   */
+  onClick?: (event: React.MouseEvent) => void;
+
+  /**
+   * Link to open when item is clicked.
+   *
+   * Absense of this prop and onClick will turn the item into non-interactive item
+   */
+  href?: string;
+
+  /**
+   * HTML target of the link
+   */
+  target?: string;
+
+  /**
+   * Item that goes on left-side of item.
+   *
+   * Will be overriden in multiselect
+   */
+  leading?: React.ReactNode;
+
+  /**
+   * Item that goes on right-side of item.
+   */
+  trailing?: React.ReactNode;
+
+  /**
+   * Item that goes immediately next to the title.
+   */
+  titleSuffix?: React.ReactElement;
+
+  /**
+   * disabled state of item
+   */
+  isDisabled?: boolean;
+
+  /**
+   * as prop to pass React Router's Link component for client-side navigation
+   */
+  as?: React.ReactComponent;
+
+  /**
+   * Color of item. set to negative for dangerous actions like Delete, Remove, etc
+   */
+  color?: 'negative';
+};
+```
+
+### Menu Subcomponent Props
+
+```ts
+// Follows the standard header, footer types of Blade like DropdownHeader, ModalHeader, etc
+type MenuHeaderProps = BaseHeaderProps;
+type MenuFooterProps = BaseFooterProps;
+
+// No props
+type MenuDividerProps = {};
+```
+
+## Example APIs
+
+### Basic API
+
+```jsx
+<Menu>
+  <Button>Edit</Button> {/* Can be Link, Avatar, or any custom interactive item */}
+  <MenuOverlay>
+    {/* Supports any JSX */}
+    <Box>Slot</Box>
+    <Divider />
+    <Box overflowY="auto">
+      <MenuItem title="Profile" />
+      <MenuItem>
+        <Text>Custom Slot in Item</Text>
+      </MenuItem>
+    </Box>
+  </MenuOverlay>
+</Menu>
+```
+
+### SubMenu
+
+```jsx
+<Menu>
+  <Button>Edit</Button>
+  <MenuOverlay>
+    <MenuItem title="Profile" />
+    {/* You can nest menu and use MenuItem as trigger for the next submenu */}
+    <Menu>
+      <MenuItem title="Accounts" />
+      <MenuOverlay>
+        <MenuItem title="Business Account" />
+        <MenuItem title="Personal Account" />
+      </MenuOverlay>
+    </Menu>
+  </MenuOverlay>
+</Menu>
+```
+
+### Custom Trigger
+
+```jsx
+<Menu>
+  <CustomComponentTrigger />
```

Comment:
Can this be used with our tooltip too? [Ref example](https://ariakit.org/examples/menu-tooltip)

Reply:
Good catch. Fixed it in office. It can be used with tooltip now by wrapping trigger with Tooltip component